File: program/src/main/java/com/oose/labsafety/user/service/UserService.java
package com.oose.labsafety.user.service;

import com.oose.labsafety.common.AbstractCatalogService;
import com.oose.labsafety.user.domain.UserAccount;
import com.oose.labsafety.user.infrastructure.UserRepository;

public final class UserService extends AbstractCatalogService<UserAccount> {

    public UserService(UserRepository repository) {
        super(repository);
    }
}

File: src/features/user/service/userService.js
import { createCrudService } from '../../../shared/service/createCrudService';
import { createUser } from '../domain/User';
import { userRepository } from '../infrastructure/userRepository';

export const userService = createCrudService({
  repository: userRepository,
  createEntity: createUser,
  idField: 'userId',
});
